tests/test_caching.py

```diff
@@ -83,3 +92,31 @@ def foo(): "deprecated since 1999"
                 with contextlib.redirect_stdout(buf):
                     run()
                 self.assertEqual(buf.getvalue(), ref)
+
+    def test_cache_dir(self):
+        fid, tmppy = tempfile.mkstemp(suffix='.py')
```

Comment:
using `tempfile.NamedTemporaryFile` with `delete=False` looks more idiomatic (and exception-safe ;-))